delta/plugins/statistics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/statistics/Statistics.scala

```diff
@@ -0,0 +1,114 @@
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClasspathResourceUtils.ioJsonObjectContentOf
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UrlUtils
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.client.{ElasticSearchClient, IndexLabel, QueryBuilder}
+import ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ElasticSearchViewRejection.WrappedElasticSearchClientError
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.model.PropertiesStatistics.propertiesDecoderFromEsAggregations
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.model.StatisticsRejection.{InvalidPropertyType, WrappedClasspathResourceError, WrappedElasticSearchRejection}
+import ch.epfl.bluebrain.nexus.delta.plugins.statistics.model.{PropertiesStatistics, StatisticsGraph, StatisticsRejection}
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.delta.sdk.Projects
+import ch.epfl.bluebrain.nexus.delta.sdk.jsonld.ExpandIri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.IdSegment
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.{ApiMappings, ProjectRef}
+import ch.epfl.bluebrain.nexus.delta.sdk.syntax._
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.ExternalIndexingConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionId.ViewProjectionId
+import io.circe.{Decoder, JsonObject}
+import monix.bio.IO
+
+trait Statistics {
+
+  /**
+    * The statistics for the relationships between different types for the passed ''projectRef''
+    */
+  def relationships(projectRef: ProjectRef): IO[StatisticsRejection, StatisticsGraph]
+
+  /**
+    * The statistics for the properties of the passed ''project'' and type ''tpe''.
+    */
+  def properties(projectRef: ProjectRef, tpe: IdSegment): IO[StatisticsRejection, PropertiesStatistics]
+
+}
+
+object Statistics {
+
+  final def apply(
+      client: ElasticSearchClient,
+      projects: Projects
+  )(implicit config: ExternalIndexingConfig): Statistics = new Statistics {
+
+    private val expandIri: ExpandIri[InvalidPropertyType] = new ExpandIri(InvalidPropertyType.apply)
+
+    implicit private val classLoader: ClassLoader = getClass.getClassLoader
+
+    private val propertiesAggQuery =
+      ioJsonObjectContentOf("elasticsearch/paths-properties-aggregations.json").memoize
+
+    private val relationshipsAggQuery =
+      ioJsonObjectContentOf("elasticsearch/paths-relationships-aggregations.json").memoize
+
+    private def propertiesAggQueryFor(tpe: Iri)                                                  =
+      propertiesAggQuery.map(_.replace("@type" -> "{{type}}", tpe))
+
+    override def relationships(projectRef: ProjectRef): IO[StatisticsRejection, StatisticsGraph] =
+      for {
+        _     <- projects.fetchActiveProject[StatisticsRejection](projectRef)
+        query <- relationshipsAggQuery.mapError(WrappedClasspathResourceError)
```

Comment:
An integration test might detect it, but simple test do not detect this kind of issues because of the nature of plugin architecture. It might not fail on unit tests but it can fail later on.

Besides that, the current behaviour is consistent with what we have in other places of the code (see `BlazegraphViewRejection`)

Reply:
There are parts where we have a different approach like in the ES plugin.
If we have a problem at runtime loading this file, the most probable is that the entire plugin has a problem and we should have a problem at startup.
It seems to me like this only complicates the rejection list

Reply:
I'm ok with doing it in another way, as long as It is consistent. I can see where we use the rejections and change it for `logAndDiscardErrors`

Reply:
Ok. I've made this change in `BlazegraphViewRejection` and `StatisticsRejection`